tests/test_pay.py

```diff
@@ -6731,3 +6731,26 @@ def test_fetchinvoice_with_payer_metadata(node_factory, bitcoind):
     inv3 = l1.rpc.call('fetchinvoice', {'offer': offer['bolt12'], 'amount_msat': '1sat', 'payer_note': 'Payment For vincenzopalazzo', 'payer_metadata': b'macros'.hex()})['invoice']
     decode3 = l1.rpc.call('decode', {'string': inv3})
     assert decode1['invreq_payer_id'] == decode3['invreq_payer_id']
+
+
+def test_pay_unannounced_routehint(node_factory, bitcoind):
+    """Tests whether sender can pay recipient through unannounced channels with
+    2 hops where the second hop uses a route hint."""
+
+    l1, l2, l3, r1, r2 = node_factory.get_nodes(5)
+
+    # l1 and l3 are connected with unannounced channels to l2
+    node_factory.join_nodes([l1, l2], announce_channels=False)
+    node_factory.join_nodes([l2, l3], announce_channels=False)
+
+    # Need these to ensure l3 adds a route hint in the invoice
+    node_factory.join_nodes([l2, r1], wait_for_announce=True)
+    node_factory.join_nodes([l2, r2], wait_for_announce=True)
+
+    # Ensure recipient sees the channels
+    wait_for(lambda: len(l3.rpc.listchannels(destination=r1.info['id'])['channels']) > 0)
+    wait_for(lambda: len(l3.rpc.listchannels(destination=r2.info['id'])['channels']) > 0)
+
+    inv = l3.rpc.invoice(2100000, 'receive', 'receive')
+    l1.rpc.pay(inv['bolt11'])
+    wait_for(lambda: l1.rpc.listpays(inv['bolt11'])['pays'][0]['status'] == 'complete')
```

Comment:
```suggestion
    result = l1.rpc.pay(inv['bolt11'])
    assert result["status"] == "complete", f"pay result is {result}"
```

This could also work, without adding `wait_for`, we should save some CPU cicly